### ISSUE TITLE
Current working directory is not used when I point to webpack.config.js in sibling directory

### ISSUE THREAD

**Issue by toni-rmc** [webpack/webpack#6865]
**Do you want to request a *feature* or report a *bug*?**

Request a feature, even though it migth be there already but I just don't know how to set it up.

**What is the current behavior?**

Basically I have `webpack.config.js` stored in different directory, not the child or parent, but sibling directory.
I'm using the `--config` switch to point to it and webpack finds it and reads from it.  
However it gives an error when this line is read for example:

`const CircularDependencyPlugin = require('circular-dependency-plugin');`

Error says that it cannot find CircularDependencyPlugin.  
It seems that webpack is looking for the `node_modules` in the directory where `webpack.config.js` is found and not in current working directory.  
So when I work in app root directory and from it invoke webpack which points to `webpack.config.js` in sibling directory seems that this other directory is used by webpack to look for the `node_modules`.  
Having duplicate `node_modules` in sibling directory is not an option.

**What is the expected behavior?**

I would like for webpack to use current working directory for `node_modules` lookup and do everything there even though I point to another `webpack.config.js` file in sibling directory with `--config` switch.  
Is there any way to set it up to behave just like in cases when `webpack.config.js` is stored in current working directory?

**If this is a feature request, what is motivation or use case for changing the behavior?**

I have multiple different applications and they all have same file structure. I also have two versions of `webpack.config.js`, one for development bundling and one for production.
I will be using `npm` to abstract webpack bundle commands. In `package.json` I want to have two commands for bundling, in each I plan to use `--config` switch to point to the prod or dev `webpack.config.js`.
Right now I have to delete and replace `webpack.config.js` in current working directory every time I want to switch from development build to production build and vice versa. And I have to do that for every separate application.
What I want to do is `cd` to root directory of the app I'm working on, each app would have `package.json` with webpack bundle commands. Those would point to dev and prod `webpack.config.js` files in sibling directory, not the parent but side by side directory with other apps directories.
This way I could pack for dev or prod without copying config files.
Due to nature of this project, yes `webpack.config.js` files (dev and prod versions) can be universal for each app.

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

Node.js 6.11.12, webpack 3.4.1, OS win 10 amd64

**Comment by ooflorent** [comment]
webpack uses node resolution algorithm. If your file is located at `./foo/bar/config.js`, the following directories will be tested:

- `./foo/bar/node_modules/`
- `./foo/node_modules/`
- `./node_modules/` (webpack is here)

**Comment by montogeek** [comment]
> It seems that webpack is looking for the node_modules in the directory where webpack.config.js is found and not in current working directory.

That sounds like a correct behaviour.

**Comment by toni-rmc** [comment]
Yes true, that is correct behaviour, but is there any way to tell webpack to use current working directory even when it comes to reading from config file in sibling directory?
So it treats that sibling directory only as a place where it reads config file from and nothing more?

**Comment by toni-rmc** [comment]
Good idea.
I'm not that experienced in webpack so I haven't thought of that nor did I know that you can merge webpack config files.
Would you use `webpack-merge` or something else?

**Comment by mcstepp** [comment]
Yes, you would use webpack-merge. Basic usage would look something like this, this would live in your root

```
var merge = require('webpack-merge)
var appConfig = require('../path/to/config') // app-specific config

var webpackConfig = merge(appConfig, {
    // your base webpack config
}

module.exports = webpackConfig
```

**Comment by toni-rmc** [comment]
I'm definitely gonna try that, thanks!
It is actually better solution than my original idea.

So basically when you do:
`var webpackConfig = merge(appConfig, {
    // your base webpack config
}`

`webpack-merge` merges all the plugins in `plugins[]` object member, as well as other entries in object.

And you can go in both directions?
Merge from `base` to `dev/prod` or from `dev/prod` to `base`.

Here is documented latter (from `dev/prod` to `common`) https://webpack.js.org/guides/production/
I need to merge from `common` to `dev/prod`.

**Comment by mcstepp** [comment]
Absolutely you can merge in either direction. You can pass in objects
directly as arguments pass in an array of objects to merge into a new
object. There's more details on it here:
https://github.com/survivejs/webpack-merge/blob/master/README.md

**Comment by toni-rmc** [comment]
Thanks.
Do you know how and which variable to set in the base config file so I can dynamically merge one or the other file, for dev or prod? Depending is the variable set for, let's say "production" or not.

**Comment by mcstepp** [comment]
This might help point you in the right direction. 

https://medium.com/@justintulk/passing-environment-variables-into-your-code-with-webpack-cab09d8974b0

**Comment by toni-rmc** [comment]
Those are useful pages. I have found `cross-env` plugin which solves some problems between Windows and POSIX variable set up.

I have split config files into `base`, `dev` and `prod` and put them all in the same (root) directory.
`base` is now included from `dev` and `prod` with `webpack-merge` and in `package.json` I have defined `build` and `build-prod` commands which use Webpack with `--config` switch so each points to the right config file.

That eliminated the need for manually copying config files every time when switching between packing options, which is what I was looking for.
Now there are two commands.

Using merge was great idea, I wouldn't come up with that by myself.
So once again, big thanks!
